tests/run-make/dump-ice-to-disk/rmake.rs

```diff
@@ -0,0 +1,55 @@
+// This test checks if internal compilation error (ICE) log files work as expected.
+// - Get the number of lines from the log files without any configuration options,
+// then check that the line count doesn't change if the backtrace gets configured to be short
+// or full.
+// - Check that disabling ICE logging results in zero files created.
+// - Check that the ICE files contain some of the expected strings.
+// See https://github.com/rust-lang/rust/pull/108714
+
+use run_make_support::{cwd, fs_wrapper, has_extension, has_prefix, rustc, shallow_find_files};
+
+fn main() {
+    rustc().input("lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    let ice_text = get_text_from_ice();
+    let default_set = ice_text.lines().count();
+    let content = ice_text;
+
+    clear_ice_files();
+    rustc().input("lib.rs").env("RUST_BACKTRACE", "short").arg("-Ztreat-err-as-bug=1").run_fail();
+    let short = get_text_from_ice().lines().count();
```

Comment:
Discussion: instead of needing explicit `clear_ice_files`, should we run each test case via `run_in_tmpdir`?

Reply:
I thought about this, but I didn't do it, because each call also sets variables that are used in the final round of assertions. In the lines you quoted, `let default_set` would have dropped out of scope before it could be compared with `short` at line 19.

Let me know if there's a rusty way to solve this I am not aware of.